src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,60 @@
+package duke;
+
+/**
+ * Class for parsing input of user for duke.Duke.
+ * @author Liew Jian Hong
+ */
+
+public class Parser {
+
+    /**
+     * Returns the result of parsing the user input in an array of string where index 0 is command, index 1 is
+     * task description and index 2 is by/at date and time.
+     *
+     * @param item User's input.
+     * @return String array of parsed item.
+     */
+    public String[] parse(String item) {
+        String[] result = new String[]{"", "", "", "false"};
+        String[] split = item.split(" ");
+        int firstWordLength = split[0].length();
+        result[0] = split[0];
+
+        String remainder = "";
+        int separator = 0;
+        if (split.length > 1) {
```

Comment:
There seem to be quite a few conditions here, maybe they could be abstracted out?